### ISSUE TITLE
[no-unused-vars] false positive for imported type alias

### ISSUE THREAD

**Issue by SleeplessByte** [typescript-eslint/typescript-eslint#363]
**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "plugins": ["@typescript-eslint"],
  "extends": [
    "plugin:@typescript-eslint/recommended",
    "plugin:react/recommended",
    "prettier",
    "prettier/@typescript-eslint",
    "prettier/babel",
    "prettier/react"
  ],
  "rules": {
    "@typescript-eslint/no-explicit-any": "off",
    "@typescript-eslint/no-parameter-properties": "off",
    "@typescript-eslint/no-unused-vars": ["on", { "argsIgnorePattern": "^_", "varsIgnorePattern": "^ignored?$" }]
  },
}
```

```TS
import { KyInput, KyOptions, unwrap } from 'use-ky'

export function observeKy<T>(input: KyInput, options: KyOptions, unwrap: unwrap<T>): Observable<T> {
  if (input === null) {
    return observableOf<T>()
  }
```

**Expected Result**

All good

**Actual Result**

```
(alias) type unwrap<T> = (a: ResponsePromise) => Promise<T>
import unwrap

'unwrap' is defined but never used.eslint(@typescript-eslint/no-unused-vars)
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.2` |
| `node`                             | `10.7.0` |
| `yarn`                              | `1.12.1` |

**Comment by davecarlson** [comment]
I can confirm this still happens in `2.0.0`:

```
import { PluginHost, CredentialProviderSource } from 'aws-cdk';

class SomeClass implements CredentialProviderSource {
  ....
}
export = {
  init (host: PluginHost) {
  }
}
```
Results in:
```
index.ts:1:10: 'PluginHost' is defined but never used.
index.ts:1:22: 'CredentialProviderSource' is defined but never used.

```

**Comment by cdalexndr** [comment]
Same issue with:
```
import DateTimeFormatOptions = Intl.DateTimeFormatOptions;
function formatDate(date: Date, options?: DateTimeFormatOptions): string {
 ....
}
```

causes:
` error  'DateTimeFormatOptions' is defined but never used  no-unused-vars`
```
λ npm ls --depth=0 | grep eslint
+-- @typescript-eslint/eslint-plugin@2.22.0
+-- @typescript-eslint/parser@2.22.0
+-- eslint@6.8.0
λ node -v
v10.16.0
```
